lab-abigail/test/food-test.js

```diff
@@ -0,0 +1,262 @@
+'use strict';
+//parse response data
+
+const server = require('../server');
+const chai = require('chai');
+const http = require('chai-http');
+const Promise = require('bluebird');
+const fs = Promise.promisifyAll(require('fs'), {suffix: 'Prom'});
+const expect = chai.expect;
+
+chai.use(http);
+
+let app;
+
+describe('HTTP Server module', function(){
+  before(function(done){
+    app = server.listen(8000);
+    done();
+  });
+
+  describe('ensure that api returns a status code of 404 for routes that have not been registered', function() {
+    it('should respond with a 404 on an invalid route', function(done) {
+      chai.request(server)
+      .get('/api/drinks')
+      .end((err, res) => {
+        expect(res).to.have.status(404);
+        done();
+      });
+    });
+  });
+
+  describe('POST method', function() {
+
+    describe('Verify item created', function() {
+      it('should create food and verify name', done => {
+        chai.request(server)
+        .post('/api/food')
+        .send({'name': 'apple', 'type': 'red', 'cost': '1.5'})
+        .end((err, res) => {
+          if (err) console.error(err);
+          expect(res.body.name).to.equal('apple');
+          done();
+        });
+      });
+
+      it('should create food and verify type', done => {
+        chai.request(server)
+        .post('/api/food')
+        .send({'name': 'apple', 'type': 'red', 'cost': '1.5'})
+        .end((err, res) => {
+          if (err) console.error(err);
+          expect(JSON.parse(res.body).type).to.equal('red');
+          done();
```

Comment:
**TODO FOR LOST POINTS**

Your expect statement fails this test. No need to parse JSON here because MongoDB is already sending you back JSON. So just make your expect statement:

```
expect(res.body.type).to.equal('red');
```